in_toto/keylib.go

```diff
@@ -152,3 +154,63 @@ func VerifySignature(key Key, sig Signature, data []byte) error {
 
 	return nil
 }
+
+/*
+ParseEd25519FromPrivateJSON parses and ed25519 private key from the json
+string. These ed25519 keys have the format as generated using in-toto-keygen:
+{
+  "keytype: "ed25519",
+  "scheme": "ed25519",
+  "keyid": ...
+  "keyid_hash_algorithms": [...]
+  "keyval": {
+    "public": "..." # 32 bytes
+    "private": "..." # 32 bytes
+  }
+}
+*/
+func ParseEd25519FromPrivateJSON(JSONString string) (Key, error) {
+
+	var keyObj Key
+	err := json.Unmarshal([]uint8(JSONString), &keyObj)
+	if err != nil {
+		return keyObj, fmt.Errorf("this is not a valid JSON key object")
+	}
+
+	if keyObj.KeyType != "ed25519" || keyObj.Scheme != "ed25519" {
+		return keyObj, fmt.Errorf("this doesn't appear to be an ed25519 key")
+	}
+
+	if keyObj.KeyVal.Private == "" {
+		return keyObj, fmt.Errorf("this key is not a private key")
+	}
+
+	// 64 hexadecimal digits => 32 bytes for the private portion of the key
+	if len(keyObj.KeyVal.Private) != 64 {
+		return keyObj, fmt.Errorf("the private field on this key is malformed")
+	}
+
+	return keyObj, nil
+}
+
+/*
+generateEd25519Signature creates an ed25519 signature using the key and the
+signable buffer provided. It returns an error if the underlying signing library
+fails.
+*/
+func generateEd25519Signature(signable []byte, key Key) (Signature, error) {
```

Comment:
We should probably also make this public, right? I.e.  `s/generateEd25519Signature/GenerateEd25519Signature/`.